modules/overrides/userconfigurable/client/limits.go

```diff
@@ -8,9 +8,9 @@ import (
 )
 
 type Limits struct {
-	Forwarders *[]string `json:"forwarders,omitempty"`
+	Forwarders *[]string `yaml:"forwarders,omitempty" json:"forwarders,omitempty"`
 
-	MetricsGenerator *LimitsMetricsGenerator `json:"metrics_generator,omitempty"`
+	MetricsGenerator LimitsMetricsGenerator `yaml:"metrics_generator,omitempty" json:"metrics_generator,omitempty"`
```

Comment:
What's the change here by becoming a non-pointer on the struct?  I'm not sure I follow.

Reply:
I did this change to simplify comparing structs. If each child-struct is a pointer it becomes nearly impossible to compare two empty structs as the pointer values will never match.

For example, the following two structs are both empty/unset:

```go
Limits {
  MetricsGenerator: &LimitsMetricsGenerator{
    DisableCollection: nil,
  },
}
```

```go
Limits {
  MetricsGenerator: nil,
}
```

But they are not equal. Having every child-struct behind a pointer creates so many variations of 'empty'.

Making them all values instead of pointers avoids this situation + it doesn't change parsing (I added a test to check).